sdk/python/kfp/dsl/_pipeline.py

```diff
@@ -331,10 +331,10 @@ def add_op(self, op: _container_op.BaseOp, define_only: bool):
             '_', '-')
         #If there is an existing op with this name then generate a new name.
         op_name = _naming._make_name_unique_by_adding_index(
-            op_name, list(self.ops.keys()), ' ')
+            op_name, list(self.ops.keys()), '-')
```

Comment:
This code hasn't been changed since it was added. Yet https://github.com/kubeflow/pipelines/issues/6433 mentioned it worked before 1.7.1. Do we know what change caused the regression?

Reply:
This issue was introduced in PR #6268
```
- self._inputs = []
+ self._inputs = self._component_spec_inputs_with_pipeline_params or []
```
The  same input is recorded both in `self._component_spec_inputs_with_pipeline_params` and `self.attrs_with_pipelineparams`,   the difference is that PipelineParam  in `self.attrs_with_pipelineparams` is santized.